### PR TITLE
Fix android builds

### PR DESCRIPTION
**Pull request by bearbin**
Currently compilation errors don't stop the build, and it generates corrupt output files.

### CODE REVIEW

File: android/compile.sh [hunk 1]
```diff
@@ -86,7 +77,7 @@ case "$1" in
 		done
 
 		cd $BASEDIR/Server
-		for file in server.zip armeabi.zip armeabi-v7a.zip arm64-v8a.zip mips.zip mips64.zip x86.zip x86_64.zip; do
```
Comment (tigerw) [inline android/compile.sh]: oof ouch my OneTouch 903 running Android 2.3.6

**Comment by bearbin** [comment]
Due to exams this will have to go on hold for a little bit as it currently doesn't compile and needs a bit of deeper thought - I should be back to this over the christmas period.

**Comment by mathiascode** [comment]
If you don't have time to fix this now, could you delete the broken Android builds from the build server, and revert to the October 2018 build? People are receiving 404 errors when downloading the binaries in the app.

**Comment by mathiascode** [comment]
I added a lazy fix. Fmt causes some issues, but I don't have the time or patience to upgrade the library right now. The Server directory could also be cleaned up a bit, but that's out of scope for this PR.

**Comment by mathiascode** [comment]
A few more things need to be done. Don't merge.

**Comment by mathiascode** [comment]
Ready for merge.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -358,7 +358,7 @@ function(make_symlink orig link)
 	# Create the symlink (platform-dependent):
 	message("Creating symlink, orig = ${orig}; link = ${link}")
 	if (CMAKE_HOST_UNIX)
-		set(command ln -s ${orig} ${link})
+		set(command ln -fs ${orig} ${link})
```
Comment (madmaxoft) [inline src/CMakeLists.txt]: This should not be needed, there's an existence check on line 354, doesn't that work?

File: src/OSSupport/Errors.cpp [hunk 3]
```diff
@@ -22,7 +22,7 @@ AString GetOSErrorString( int a_ErrNo)
 
 	// According to https://linux.die.net/man/3/strerror_r there are two versions of strerror_r():
 
-	#if defined(__GLIBC__) && defined( _GNU_SOURCE) && !defined(ANDROID)  // GNU version of strerror_r()
+	#if defined(__GLIBC__) && defined( _GNU_SOURCE) || defined(ANDROID)  // GNU version of strerror_r()
```
Comment (madmaxoft) [inline src/OSSupport/Errors.cpp]: This, for me, is now unreadable. Does the `&&` have priority over `||`, or are they evaluated left-to-right? Please parenthesize properly.

**Comment by madmaxoft** [comment]
(Sorry for breaking my comments into little batches, I usually re-read PRs multiple times and each time I notice something new.)

**Comment by mathiascode** [comment]
No worries!

**Comment by mathiascode** [comment]
If this is merged, don't remove the androidbuild branch until bearbin points the Android builds to the master branch again.

**Comment by mathiascode** [comment]
There are some issues with the builds on older Android versions. Investigating.

**Comment by mathiascode** [comment]
Everything is fine now. The server starts properly on old devices.

**Pull request merged**